backend/src/search/tests/search.service.mock.ts

```diff
@@ -0,0 +1,71 @@
+import mockCrossRefData from './crossref.mock';
+import mockSearchData from './searchdata.mock';
+import { HttpService } from '@nestjs/axios';
+import { CrossRefPaperResponse, CrossRefResponse, PaperInfo } from '../entities/crossRef.entity';
+
+export function mockHttpService() {
+  const httpService = new HttpService();
+  jest.spyOn(httpService['axiosRef'], 'get').mockImplementation((url: string) => {
+    const params = new URLSearchParams(new URL(url).search);
+    const keyword = params.get('query');
+    if (keyword === null) {
+      const item = mockCrossRefData.message.items[0];
+      return Promise.resolve<{ data: CrossRefPaperResponse }>({
+        data: { message: item },
+      });
+    }
+    const rows = parseInt(params.get('rows'));
+    const selects = params.get('select').split(',');
+    const items = mockCrossRefData.message.items.slice(0, rows).map((item, i) => {
+      return selects.reduce((acc, select) => {
+        if (select === 'title') item[select] = [`${keyword}-${i}`];
+        acc[select] = item[select];
+        return acc;
+      }, {});
+    });
+
+    return Promise.resolve<{ data: CrossRefResponse }>({
+      data: { message: { 'total-results': mockCrossRefData.message['total-results'], items } },
+    });
+  });
+  return httpService;
+}
+
+export function mockElasticService() {
+  // TODO: should mockup index?
+  const index = jest.fn().mockResolvedValue(() => {
+    return true;
+  });
+  const search = jest.fn();
+  search.mockResolvedValueOnce({
+    hits: {
+      total: {
+        value: 0,
+      },
+    },
+  });
+  search.mockResolvedValue({
```

Comment:
mockResolvedValueOnce로 호출 횟수에따라 케이스를 구분해서 테스트에 활용할 수 있었군요! 좋은 방법을 찾아서 적용하신 것 같아요. 배워갑니다.

```suggestion
  })
    .mockResolvedValue({
```

이부분 체이닝이 가능하더라고요. jest 공식 docs 예제에서도 저 메서드를 사용할 때 체이닝 패턴을 주로 사용 하던데, 개인적으로는 체이닝을 하면 이런 순서로 호출될것이다~ 라는 느낌을 주는 것 같습니다..!?!! (물론 순서가 바뀌더라도 mockResolvedValueOnce가 항상 먼저 불리긴합니다..) 사소한부분이라... 성빈님의 취향에 맡기겠습니다.

Reply:
체이닝 되는 게 가독성이 더 좋은 것 같습니다!